sound/soc/sof/intel/hda-codec.c

```diff
@@ -151,6 +151,8 @@ static int hda_codec_probe(struct snd_sof_dev *sdev, int address,
 		if (!hdev->bus->audio_component) {
 			dev_dbg(sdev->dev,
 				"iDisp hw present but no driver\n");
+
+			snd_hdac_ext_bus_device_exit(hdev);
```

Comment:
maybe you want to consider making a common error handling exit path in this function and use a `goto`

Reply:
thanks